subsys/bluetooth/mesh/gen_ponoff_srv.c

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <bluetooth/mesh/gen_ponoff_srv.h>
+
+#include <string.h>
+#include <settings/settings.h>
+#include <stdlib.h>
+#include "model_utils.h"
+
+/** Persistent storage handling */
+struct ponoff_settings_data {
+	u8_t on_power_up;
+	bool on_off;
+} __packed;
+
+static int store(struct gen_ponoff_srv *srv,
+		 const struct gen_onoff_status *onoff_status)
+{
+	struct ponoff_settings_data data;
+
+	data.on_power_up = (u8_t)srv->on_power_up;
+
+	switch (srv->on_power_up) {
+	case ON_POWER_UP_OFF:
+		data.on_off = false;
```

Comment:
Yeah, this connection is a bit awkward. The problem is that we need the last known value of the onoff server on bootup, which we don't really expect it to know, since that's this model's job. We're currently intercepting the callbacks for the generic onoff server and acting on those without duplicating the state in RAM. I couldn't find a better way to solve this, but I'm open to suggestions.